### PR TITLE
Scroll to top fix 143

### LINKED ISSUE #143: When browsing feed history, user lands at the bottom of the feed
0. Pre-Condition: enough posts in homefeed for Older Items link to appear.
1. Scroll down and click Older Items link -> you are on ?offset=25 page.

Expected: user is at the top of the page and can read all items.
Actual: user lands at the bottom of the page and has to scroll up.

### PR DESCRIPTION
**Pull request by innin**
fixes #143 : scroll to top after route transition and when `afterModel` is triggered. Also added css class `.transition-active` to body element when route transition is active (between `beforeModel` and `afterModel`) this allows to create simple loader indicator

### CODE REVIEW

File: themes/helvetica/base.scss [hunk 1]
```diff
@@ -2,6 +2,20 @@ body {
...
...I8qhEolI5Nhg5FJ/zBt2nNmVKlCBXzUk7RVk2JbuVpkiiMoWDw7/d24VJZV3I9rmXqjB9ZAgFcMTxwoKyctQrUY0+o41/cew05lW5kyfPip4Ruk+hY4odCbtLK2qjb0RjpjYsOD+hogoZFV2Ut3/Z1QxBgP54MGYzc8ca1vLJlXyy5leKOSAgfIHZTxuPPGqS3jmGZR7eLgn7IQbitsgC1DcuQZGnT/ljtgWIIxlKLvRtN30UFAZTGgvGGpwPfk+HCINpSgf38Ms5zSkAh9kZLQA2VlotAJ23h3H3JoaHdQPf5VyMo0Am7nllBDKOTETcNkqI8UDyangA4NXVDhiqwMgNZBRtTilwQopfPHBVZkOMQxJTqQgR0SKlTIikSa0U2G/rTwwkwTFWNDjwoMops/D2BA5IqwUXYQAyXGY5thp7gxxpX+aaGlP3w8SJRoaZ2gxypkPnCmN9ckF88iQWp5ypsrLqDLmf6cso0m10wxJWWnoEJAf+gwAahDN7D56EI3PPIAHJNmqimgLRhhhAB8rLDFCaU9KimgMAyh6qqrquBCFAK8//iRNycYQYIXH4DQgqa9sOorq1lssRCttnoBgAfIIhvDko+y8OuzqhZ1gw63HpvstchGkakL0P7qwg3CYCtusgxkukWv3a4aBQzjjkuGrM2uwIcAUTDgQha+ulDMFO1i+4ERm870whYtwLACC/6w2y4ZUEygz6lprcMFF+sA2si1DE8gwAmivfCAMYUY9ogPJPsQspbhRMGIIxuf+gI4FMQ8S1ohlEyyFocC6g0SMfdcsUbhLISEzTfnrOUDPfcsxkHFCLMGLL0oRAnRJF9hdFqcEJB0zD9TMcPXM1Tnjy5DU/1IwC90sDUFAxx0Cthfa6CQO2XbjIQSmnozy9oWZNpYDNwzwFIVJ3WXfHemhWidNAFnH3QD4DMg6c86hZOMBCd4MaULIWv/gZI3fQCOA0NKVO4DJQ2dxpA3cqwth3oOSQD4Cg2VTjQSqauuUARrEyIsQ0EAzsdHI9uMOk+Wdbb34mYOCLhYH11RNhI/J5SXQ0gnPcupwgBODVPuRBDBnwqtpFDaPZvAbENZAF7umU4d5s8LTAzABcIzuQC42GlZ71RuZ5oC4LJgGP/9D4CUMQLgVECZAxIkI2k5hey+toZdpcWBFdHSK6iwDXEMbysYNN9WvBGpQx0wIAA7)
```
Comment (epicmonkey) [inline themes/helvetica/base.scss]: Any objections to extract this image to a static file that clients/browsers will cache?

**Comment by epicmonkey** [comment]
Does it make sense to mix App.TransitionalRoute into the following routes as well:
- routes/GroupsHomeRoute
- routes/GroupsNewRoute
- routes/SessionNewRoute
- routes/SessionDestroyRoute
- routes/UsersNewRoute

File: themes/helvetica/base.scss (hunk 1 above)
Comment (clbn) [inline themes/helvetica/base.scss]: BTW, it looks like `/img/uploading.gif`.

**Comment by innin** [comment]
@clbn yes, that's `img/uploading.gif` )
@epicmonkey No objections but usually I prefer `data:uri` instead of direct image load for small images because it's rendered faster (css is being cached as well). But there's a problem: even if I will replace it, I'm not happy in general with animated gif in this case, because it freezes sometimes (looks like JS code is blocking browser at some moments during transition. It might be synchronous xhr call or some long  loop). I will try css-animated loader and if it won't freeze I think it will look better.

**Comment by innin** [comment]
@epicmonkey I will add this mixin to `UsersNewRoute`, `GroupsHomeRoute ` and `GroupsNewRoute` but I'm not sure if it makes any sense to add it to `SessionNewRoute` and `SessionDestroyRoute`: this ones just make redirects

**Comment by innin** [comment]
Updated this pr.
css-animation also freezes, so I just replaced data:uri with `/img/uploading.gif`
And added mixin to missing routes

File: public/js/app/components/TransitionalRoute.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    beforeModel: function() {
+      this._super();
+      Ember.$('body').addClass('transition-active');
+    },
+
+    afterModel: function() {
+      this._super();
+      window.scrollTo(0,0);
+      Ember.$('body').removeClass('transition-active');
+    }
```
Comment (clbn) [inline public/js/app/components/TransitionalRoute.js]: Some nitpicking about code style: semicolons! I love them, but...

File: public/js/app/routes/PostRoute.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 define(["app/app",
-        "ember"], function(App, Emberx) {
+        "ember", "components/TransitionalRoute"], function(App, Ember) {
```
Comment (clbn) [inline public/js/app/routes/PostRoute.js]: More nitpicking on code style: maybe it's better to put every dependency on a new line?

**Comment by clbn** [comment]
There are some comments. Except for that: :+1:

**Comment by innin** [comment]
@clbn updated
P.S. I cannot  stop using semicolons after 10 yrs of Perl/PHP development :)

**Comment by clbn** [comment]
Great commit! I tried really hard, but the only issue I found is indentation on 2nd line of `TransitionalRoute.js`.

No, wait, there is something for *unauthenticated* visitors, check it out:

1. Go to someone's page: http://frf.creagenics.com:3333/th
2. Click on "logo" ("Pepyatka")
3. You'll get infinite throbber

Seems like some routes are not covered, eh?

**Comment by innin** [comment]
Pushed update. Thanks for testing this! My bad, I totally forgot to test with not logged user. @epicmonkey was right: `Session*` routes need this mixin as well

**Comment by clbn** [comment]
:+1:

**Pull request merged**